tool/release/createnotes/CommitDescription.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.tool.release.createnotes
+
+import com.eclipsesource.json.Json
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelBug
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelFeature
+import com.vaticle.dependencies.tool.release.createnotes.Constant.github
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelPrefix
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelRefactor
+
+data class CommitDescription(val title: String, val desc: String, val type: Type) {
+    enum class Type { FEATURE, BUG, REFACTOR, OTHER }
+}
+
+fun getCommitDescriptions(org: String, repo: String, commits: List<String>, githubToken: String): List<CommitDescription> {
```

Comment:
this function collects the commit description from the commit:
- if the commit is created from a PR, the title and description will be used.
- if the commit is directly pushed, the first line will be collected and the rest discarded. commit message may be less structured than a PR message and we do not want to accidentally add implementation detail.

Reply:
now that I think about it, it's better to filter out the "implementation" section the PR description here